sdx-submitter.go

```diff
@@ -0,0 +1,168 @@
+package main
+import(
+		"fmt"
+		"flag"
+		"github.com/streadway/amqp"
+		"os"
+		"io/ioutil"
+)
+
+func main() {
+
+	// Read a source file and place it on a rabbit topic exchange
+	var url, exchange, routing_key,    encryption_key,    signing_key, name, password string
+	var port, host , queue, vhost, message_path_and_file string
+	var msg_body []byte
+	var err error
+
+	getArguments(&name, &password, &port, &host, &url, &queue, &exchange, &routing_key, &encryption_key, &signing_key, &vhost, &message_path_and_file)
+
+	msg_body = getBody(message_path_and_file)
+
+	// If encyrpt specified then encrypt
+	// if sign specified then sign
+
+	sendToRabbit(url, exchange, queue, routing_key, err, msg_body)
+	
+}
+
+func getArguments(name *string,
+	password *string,
+	port *string,
+	host *string,
+	url *string,
+	queue *string,
+	exchange *string,
+	routing_key *string,
+	encryption_key *string,
+	signing_key *string,
+	vhost *string,
+	message_path_and_file *string)  {
+	// define the command line parameters
+	flag.StringVar(name, "n", "", "name of the rabbit user")
+	flag.StringVar(password, "p", "", "password of the rabbit user")
+	flag.StringVar(port, "rport", "", "port used to connect to rabbit")
+	flag.StringVar(host, "rhost", "localhost", "hostname used to connect to rabbit")
+	flag.StringVar(vhost, "v", "", "vhostname used to connect to rabbit")
+	flag.StringVar(url, "u", "", "url connection string ")
+	flag.StringVar(queue, "q", "", "name of the rabbit queue")
+	flag.StringVar(exchange, "x", "message", "name of the rabbit exchange")
+	flag.StringVar(routing_key, "r", "sdx-made-up-routing-key", "rabbit routing key")
+	flag.StringVar(encryption_key, "e", "sdx-made-encryption_key", "path to a private key file used for encryption")
+	flag.StringVar(signing_key, "s", "sdx-made-signing_key", "path to a private key used for signing")
+	flag.StringVar(message_path_and_file, "f", "", "path to filename to send")
+	flag.Parse()
+	// Use values from environment variables if no value empty
+	getFromEnvIfEmpty(name, "RABBITMQ_DEFAULT_USER", "guest")
```

Comment:
Best to avoid defaults - don't allow the program to continue if it's not configured. That way you avoid the situation where you accidentally start publishing messages to the wrong queue because it's falling back to a default when you think you set the value.

Reply:
with the exception of the routing key , source file , signing key file and encryption key file , all values have hardcoded values . Possibly routing key needs to be defaulted also ?